src/main/scala/is/hail/annotations/Region.scala

```diff
@@ -149,10 +155,33 @@ final class Region(
 
   private def allocate(n: Long): Long = {
     assert(n >= 0)
-    val off = end
-    ensure(n)
-    end += n
-    off
+    if (n > blockSize) {
+      // FIXME: is this guaranteed to be aligned to anything?
```

Comment:
malloc should return a pointer "suitably aligned for any object type".

I have played games in the past with writing an allocator which would return
8byte aligned addresses if the size was a multiple of 8, but 4byte-aligned addresses
if the size was not a multiple of 8.

On the whole I would trust malloc to give 8byte alignment, but if I needed bigger
alignment then I would make it explicit by using "void *memalign(size_t alignment, size_t size);"

But that's just paranoia and rumors, not based on definite documentation.